src/Properties/ModelPropertyExtension.php

```diff
@@ -132,6 +134,13 @@ public function getProperty(
             throw new ShouldNotHappenException();
         }
 
+        if (version_compare(LARAVEL_VERSION, '11.0.0', '>=')) {
+            // Needed to merge the model casts method into the property
+            // @phpstan-ignore-next-line
+            $reflection = new ReflectionMethod($modelInstance, 'initializeHasAttributes');
+            $reflection->invoke($modelInstance);
```

Comment:
This is not ideal. We should not add new code that actually executes the analyzed code. I know there are codes like that already, but the least we could do is to avoid adding new ones.

We might check if the `casts` method has return type with array shape maybe. If yes we can use that directly! If not we need to parse the file and analyze it's contents to understand the shape it returns. But I hope most of the developers will write the array shape return type

Reply:
I know it's not ideal, but the whole reason that Laravel included the `casts()` method was so that you could have more complicated expressions (like function calls) in the cast definition---instead of trying to string concatenate a class name with argument together:

```php
/**
 * Get the attributes that should be cast.
 *
 * @return array<string, string>
 */
protected function casts(): array
{
    return [
        'options' => AsCollection::using(OptionCollection::class),
    ];
}
```

Instead of:

```php

protected $casts = [
        'options' => AsCollection::class.':'.OptionCollection::class,
];
```

Behind the scenes, everything gets compiled down to `array<string, string>` so that it can be merged into the `$casts` property (that Larastan understands), but the function needs to be called to do that. I'm not sure how well Larastan will be able parse more complicated expression.

Thought?

Reply:
Also, the more I think about this I come to the conclusion that developers should provide the array shape and we read that. We don't even need to parse the file.

Reply:
> Then I would say don't do stupid things inside the casts method? 🙃

Exactly. But people will do that and they will come here to complain about it. Happened in past.

> However, this would fail phpstan because array{option: string} is actually returned...which is not helpful for us to use

Yeah, I don't have a solution for that now.. But it's only an issue with the `using` method

Reply:
```php
* @return array{
 *     options: 'OptionCollection',
 *     started_at: 'Carbon',
 *     is_complete: 'bool',
 *     status: 'int',
 *     total: 'float',
 * }
 ```

this is what I meant. Just like `$casts` property. And we handle that currently by mapping these string to PHPStan type.

Reply:
I'm still not sure how viable that is given that the `casts` method was created due to the headaches of trying to define more complicated types solely by strings / string concatenation.

Not sure many people would care for the duplication either---if I can describe the cast by a simple string there's there's no reason for me to create the `casts` method with the `@return` doc when I can just use the `$casts` property

Reply:
> if I can describe the cast by a simple string there's no reason for me to create the casts method

Great. So you can keep using `$casts` property. It's not deprecated.

Reply:
While the `$casts` property might not be deprecated, it's no longer documented in the latest documentation (https://laravel.com/docs/11.x/eloquent-mutators#attribute-casting). It seems somewhat paradoxical to prioritize safeguarding developers from potential pitfalls while simultaneously expecting them to be familiar with previous framework versions.

Reply:
I agree with "how about we release this to support L11" since it was released almost 2 months ago and this is breaking compatibility for all users...

Reply:
I cannot agree more: all my CIs are red for 2 months for this specific problem, and I believe that many people are in my situation

Reply:
These errors are not a blocker. You can ignore them and make your build green. Larastan supports L11. It is just missing a feature. It's not urgent or anything.

And like I said, I cannot merge this implementation.